src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1486,37 +1488,49 @@ public void setAllowFlight(boolean flight)
 	@Deprecated
 	public void hidePlayer(@NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		hidePlayer("", player);
 	}
 
 	@Override
 	public void hidePlayer(@NotNull Plugin plugin, @NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		hidePlayer(plugin.getName(), player);
+	}
+
+	private void hidePlayer(@NotNull String pluginName, @NotNull Player player)
+	{
+		hiddenPlayers.putIfAbsent(player.getUniqueId(), new ArrayList<>());
+		List<String> blockingPlugins = hiddenPlayers.get(player.getUniqueId());
+		if(blockingPlugins.contains(pluginName)) return;
+		blockingPlugins.add(pluginName);
 	}
 
 	@Override
 	@Deprecated
 	public void showPlayer(@NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		showPlayer("", player);
 	}
 
 	@Override
 	public void showPlayer(@NotNull Plugin plugin, @NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		showPlayer(plugin.getName(), player);
+	}
+
+	private void showPlayer(@NotNull String pluginName, @NotNull Player player)
+	{
+		if(!hiddenPlayers.containsKey(player.getUniqueId())) return;
+		List<String> blockingPlugins = hiddenPlayers.get(player.getUniqueId());
+		blockingPlugins.remove(pluginName);
+		if(!blockingPlugins.isEmpty()) return;
+		hiddenPlayers.remove(player.getUniqueId());
```

Comment:
Is it a good idea to use `pluginName` here? Wouldn't it be better to simply use the `Plugin` itself as the key?

```suggestion
		if (hiddenPlayers.containsKey(player.getUniqueId()))
		{
			List<String> blockingPlugins = hiddenPlayers.get(player.getUniqueId());
			blockingPlugins.remove(pluginName);
			if (blockingPlugins.isEmpty())
				hiddenPlayers.remove(player.getUniqueId());
		}
```